tensorflow/python/ops/math_ops.py

```diff
@@ -1771,6 +1776,8 @@ def matmul(a,
           a, b, transpose_a=transpose_a, transpose_b=transpose_b, name=name)
 
 
+_OverrideBinaryOperatorHelper(matmul, "matmul")
```

Comment:
I did not add `if sys.version_info >= (3, 5)` because it probably does not hurt to add `__matmul__()` and `__rmatmul__()`, and it simplifies the code and the tests.